apps/communications/contacts/js/contacts.js

```diff
@@ -691,6 +670,29 @@ var Contacts = (function() {
     window.removeEventListener('DOMContentLoaded', onLoad);
   });
 
+  window.addEventListener('pageshow', function onPageshow() {
+    // #new handling
+    var contactID = sessionStorage.getItem('contactID');
+    var reason = sessionStorage.getItem('reason');
+    if (!contactID || (contactID.length && contactID.length ===0)) {
+      var pendingOp = sessionStorage.getItem('oncontactchange');
+
+      // Invoke oncontactchange manually
+      if (typeof pendingOp !== 'undefined') {
+        oncontactchange(JSON.parse(pendingOp));
+        sessionStorage.setItem('oncontactchange', null);
+        return;
+      }
+      return;
+    }
+    performOnContactChange({
+      contactID: contactID,
+      reason: reason
+    });
+    sessionStorage.setItem('contactID', null);
+    sessionStorage.setItem('reason', null);
+  });
```

Comment:
Wouldn't it better to save in sessionStorage a stringified object? Something like this:

```javascript
var evt = {
  contactID: id,
  reason: 'foo'
};

sessionStorage.setItem('oncontactchange', JSON.stringify(evt));
```

An then in this handler we just need to check if the key `onconctactchange` exists. Besides, this code breaks the UI if `contactID` and `reason` are `undefined`, because performOnContactChange is called with an undefined argument and this function tries to access to the contactID property of the event, causing a JS error. (That was the reason why headers were broken in the first demo we did.